lib/AST/ASTImporter.cpp

```diff
@@ -282,6 +282,34 @@ namespace clang {
         ToD->setImplicit();
     }
 
+    void addDeclToContexts(Decl *FromD, Decl *ToD) {
+      if (!Importer.isMinimalImport()) {
+        bool Added = false;
+        if (FromD->getDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (ToD->getDeclContext() != ToD->getLexicalDeclContext() &&
+            FromD->getLexicalDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getLexicalDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (auto *FromTD = dyn_cast<TagDecl>(FromD)) {
```

Comment:
This block might be better to put into a separate function like `makeDeclVisible(Decl *FromD, Decl *ToD)` and called only from `VisitRecordDecl`?